contracts/lib/SafeMathInt.sol

```diff
@@ -81,4 +81,44 @@ library SafeMathInt {
         require(a != MIN_INT256);
         return a < 0 ? -a : a;
     }
+
+    /**
+     * @dev Computes 2^exp with limited precision where -100 <= exp <= 100 * one
+     * @param one 1.0 represented in the same fixed point number format as exp
+     * @param exp The power to raise 2 to 0 <= exp <= 100 * one
+     * @return 2^exp represented with same number of decimals after the point as one
+     */
+    function twoPower(int256 exp, int256 one) internal pure returns (int256) {
+        bool reciprocal = false;
+        if (exp < 0) {
+            reciprocal = true;
+            exp = abs(exp);
+        }
+
+        // Precomputed values for 2^(1/2^i) in 18 decimals fixed point numbers
+        int256[5] memory ks = [
+            int256(1414213562373095049),
+            1189207115002721067,
```

Comment:
I'm curious why the first element needs explicit type cast but the subsequent ones don't. Does the compiler infer because all array elements expected to be the same?

Reply:
Yes, that's the case.